src/core/CPU.cpp

```diff
@@ -1182,7 +1169,16 @@ Blaze::Cycles Blaze::CPU::executeCPY(AddressingMode mode) {
 };
 
 Blaze::Cycles Blaze::CPU::executeDEC(AddressingMode mode) {
-	// TODO
+	// TODO TO CHECK
+	Address addr = loadOperand(mode);
+	A--;
+	if (memoryAndAccumulatorAre8Bit()) {
+		store8(addr, A.load());
+	}
+	else {
+		store16(addr, A.load());
+	}
```

Comment:
Actually, what `DEC` does is that it loads the memory address, subtracts 1 from the loaded value, and then stores this value back to the same memory address. For this instruction, you're going to want to use `decodeAddress` instead of `loadOperand`: `decodeAddress` gives you the operand address, which you can then load from and store to, while `loadOperand` gives you the value at the operand address (with no way to store back to it).